08-graphs/02-graphs-algorithms.md

```diff
@@ -240,21 +569,20 @@ In the worst-case you will need to add each node to the Stack, so the space comp
-
-- Detecting a cycle in a graph
-- Finding a path in a maze where there is only one correct path
-- Scheduling jobs based on dependencies on other jobs
-
 
 ## Summary
 
-In this lesson we have looked at the Graph data structure.  A graph is a collection of nodes and connections between nodes called edges.  Binary search trees and linked lists are both subsets of graphs.  A graph can have both unweighted and weighted edges.  In a weighted graph, each connection is assigned a cost or weight.  Further a graph can be directed or undirected.  In a directed graph an edge is not bidirectional.  
+In this lesson we have looked at the Graph data structure. A graph is a collection of `nodes` and the connections between nodes are called `edges`. Binary search trees and linked lists are both subsets of graphs.  
+
+Here are some other facts about graphs:
+- A graph can have both unweighted and weighted edges.  In a weighted graph, each connection is assigned a cost or weight. 
+- A graph can be directed or undirected.  In a directed graph an edge is *not* bidirectional.  
 
-There are two main ways to represent a graph.  The first is using an adjacency matrix which is a 2-dimensional array where each entry represents an edge between the nodes indicated by it's row and column indices.  An adjacency matrix is very quick to lookup a connection between two nodes.  On the other hand retrieving a list of all the edges a node has a time complexity of O(N) where `N` is the number of nodes.  It also require O(N<sup>2</sup>) space complexity.  The other method of representing a graph involves an adjacency list, which is an array of lists of neighbors.  This method has an O(1) time complexity to look up all the neighbors of a node, but O(d) to lookup a particular connection where `d` is the degree of the node, or the number of neighbors it has.  It also requires less space complexity.
+There are a few main ways to represent a graph:
+1. Using an adjacency dictionary, which represents the nodes as keys and the edges are represented in a list as the value of the key. This method has an O(1) time complexity to look up the neighbors of the node. We used this representation in the challenges above.
+1. Using an adjacency matrix which is a 2-dimensional array where each entry represents an edge between the nodes indicated by it's row and column indices.  An adjacency matrix is very quick to lookup a connection between two nodes.  On the other hand retrieving a list of all the edges a node has has a time complexity of O(N) where `N` is the number of nodes.  It also require O(N<sup>2</sup>) space complexity.  
+1. Another method of representing a graph involves an adjacency list, which is an array of lists of neighbors.  This method has an O(1) time complexity to look up all the neighbors of a node, but O(d) to lookup a particular connection where `d` is the degree of the node, or the number of neighbors it has.  It also requires less space complexity.
```

Comment:
I really like these explanations, but I would consider moving them to the overview lesson as it matches the content of that lesson more closely. 

Here I might consider going more in depth about why BFS and DFS lend themselves to queues v stacks respectively and situations in which we might we might use each or choose one over the other, especially given they have the same Big O.